components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java

```diff
@@ -440,15 +445,34 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIManagem
     @Override
     public List<Label> getLabelInfo(List<String> labels) throws APIManagementException {
 
-        List<Label> labelList;
+        List<Label> filteredLabels;
+        String labelExtractorClassName = config.getLabelExtractor();
         try {
-            labelList = getLabelDAO().getLabelsByName(labels);
+            List<Label> availableLabels = getLabelDAO().getLabelsByName(labels);
+
+            // TODO: Need to update this based on the roles impl.
+            // Logged in user
+            String user = "admin";
```

Comment:
Used hard-coded values as we have hard coded users at the moment. Based on a discussion, this implementation has changed and roles will not be used. Username of the logged in user will be used.